packages/hooks/src/useCreateOffer.ts

```diff
@@ -53,6 +33,42 @@ gql`
   }
 `
 
+gql`
+  mutation CreateCurrencyApprovalTransaction(
+    $accountAddress: Address!
+    $amount: Uint256!
+    $currencyId: String!
+  ) {
+    createCurrencyApprovalTransaction(
+      accountAddress: $accountAddress
+      amount: $amount
+      currencyId: $currencyId
+    ) {
+      transaction {
+        ...Transaction
+      }
+    }
+  }
+`
+
+gql`
+  mutation CreateCollectionApprovalTransaction(
+    $accountAddress: Address!
+    $chainId: Int!
+    $collectionAddress: String!
+  ) {
+    createCollectionApprovalTransaction(
+      accountAddress: $accountAddress
+      chainId: $chainId
+      collectionAddress: $collectionAddress
+    ) {
+      transaction {
+        ...Transaction
+      }
+    }
+  }
+`
```

Comment:
Now that these are dedicated mutations, instead of creating the queries in `useCreateOffer` and using it in both `useCreateOffer` and `useAcceptOffer`, let's instead have a dedicated (of even 2) hooks for approval that `useCreateOffer` and `useAcceptOffer` will use.

`useApproveCollection` and `useApproveCurrency`

Reply:
Added but I need to properly test this.
The code is a lot more complex at the end...

Reply:
Tested with the currency approval. The collection approval is a lot more tricky to test because the ERC contract we deploy using the rarible token deployer approve automatically the proxies.